django/db/models/base.py

```diff
@@ -450,6 +450,14 @@ def __init__(self, db=None):
         # Necessary for correct validation of new instances of objects with explicit (non-auto) PKs.
         # This impacts validation only; it has no effect on the actual save.
         self.adding = True
+        self.cache = {}
+
+    def set_fields_cache(self, field, value):
```

Comment:
Yeah, I think that falls into YAGNI territory. Better to just do the simpler and faster thing for now, and add indirection later if/when we actually need it.